.travis.yml

```diff
@@ -52,23 +52,21 @@ jobs:
     # This tests the formatting of a PR.
     - if: (type = pull_request) OR (type = cron) OR NOT (branch = master)
       stage: test
-      name: "pgindent and license check"
+      name: "clang-format and license check"
       env:
-        - PG_VERSION=10.2 PG_GIT_TAG=REL_10_2 ADDITIONAL_CMAKE_FLAGS="-DUSE_DEFAULT_VISIBILITY=1"
+        - PG_VERSION=11.1 ASAN_OPTIONS="detect_leaks=0"
       before_install:
-        - git clone --branch ${PG_GIT_TAG} --depth 1 https://github.com/postgres/postgres.git /tmp/postgres
-        - docker run -d --name pgbuild -v ${TRAVIS_BUILD_DIR}:/build -v /tmp/postgres:/postgres timescaledev/pgindent:10-alpine
+        - docker run -d --name pgbuild -v ${TRAVIS_BUILD_DIR}:/timescaledb timescaledev/postgres-dev-clang:clang7-pg${PG_VERSION} /bin/sleep infinity
```

Comment:
1. what does this custom image do over just providing `clang-format`?
2. isn't there a `deb` we can install on the build host (avoiding Docker)
3. Why `clang-format` 7 and not 8? Is v7 a requirement?
4. If `clang-format` 7 is indeed a requirement, should we enforce it also for local installs? (after a `find_program` check)

Reply:
1. The custom image has clang and postgres compiled/installed with clang with sanitizers turned on. So all the clang tools (clang-tidy etc.) are there  as well as the ability to compile timescaledb and run tests under a clang compilation. This will be useful for running tests under sanitizers that can check for memory leaks/misalignment/etc.
2. Since different devs use different oses debs won't work. Docker is the easiest cross platform thing to use.
3. 8 isn't stable. 7 is the latest stable and is a requirement.
4. will do.

Reply:
FWIW, since this is a Travis build configuration and not a generic Makefile or similar I don't see why (2) would be an issue.